pkg/sql/sem/builtins/builtins.go

```diff
@@ -4927,6 +4928,32 @@ may increase either contention or retry errors, or both.`,
 		},
 	),
 
+	"crdb_internal.version_upgrade_type": makeBuiltin(
+		tree.FunctionProperties{Category: categorySystemInfo},
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"to_version", types.String},
+			},
+			ReturnType: tree.FixedReturnType(types.String),
+			Fn: func(evalCtx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				toVersion, err := version.Parse(string(tree.MustBeDString(args[0])))
+				if err != nil {
+					return nil, pgerror.New(pgcode.InvalidParameterValue, "unable to parse version")
+				}
+
+				currentVersion := version.MustParse(build.GetInfo().Tag)
```

Comment:
This is purely local to the node, but what about mixed-version clusters? Say you have two nodes running the 21.1 binary and one node still on 20.2. If you ask 21.1 for moving to 21.1.5 it will tell you it's a minor upgrade, if you ask the 20.2 node it tells you major upgrade, etc. How do you want this to be handled? Is purely local enough?

Do you perhaps want a pure builtin that takes a `(current_version, target_version)` tuple and then you call it with `crdb_internal.node_executable_version()` as the first parameter, or something like `select crdb_internal.version_upgrade_type(server_version, '20.1.5')` from crdb_internal.kv_node_status`?

But note that there are two kinds of versions in CRDB, the internal cluster versions (`20.2-N`) and the calver external ones (20.2.5). CRDB internally has on way to distinguish patch releases, except through the build tag, which prior to this change has no semantics attached to it.